server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1859,12 +1859,30 @@ public void close(String reason, boolean flushEngine) throws IOException {
                     // playing safe here and close the engine even if the above succeeds - close can be called multiple times
                     // Also closing refreshListeners to prevent us from accumulating any more listeners
                     IOUtils.close(engine, globalCheckpointListeners, refreshListeners, pendingReplicationActions);
+
+                    // Remote Segment Store and Remote Translog Cleanup
+                    if (deleted && isPrimaryMode() && isRemoteStoreEnabled()) {
+                        RemoteSegmentStoreDirectory remoteDirectory = getRemoteDirectory();
+                        remoteDirectory.deleteStaleSegments(0);
+                        remoteDirectory.deleteIfEmpty();
+                        // Translog Clean up
+                        engine.translogManager().onDelete();
```

Comment:
`isRemoteStoreEnabled()` will not check for remote translog. We need separate check

Reply:
translog clean up is independent of remote. Moved it out of `isRemoteStoreEnabled` .